jsonnet/components/metrics-server.libsonnet

```diff
@@ -0,0 +1,374 @@
+      name: 'metrics-server',
+      namespace: cfg.namespace,
+    },
+    spec: {
+      replicas: 2,
+      selector: {
+        matchLabels: {
+          'app.kubernetes.io/name': 'metrics-server',
+        },
+      },
+      strategy: {
+        rollingUpdate: {
+          maxUnavailable: 1,
+        },
+      },
+      template: {
+        metadata: {
+          labels: {
+            'app.kubernetes.io/name': 'metrics-server',
+            'app.kubernetes.io/component': 'metrics-server',
+          } + cfg.commonLabels,
+        },
+        spec: {
+          affinity: {
+            podAntiAffinity: {
+              requiredDuringSchedulingIgnoredDuringExecution: [
+                {
+                  labelSelector: {
+                    matchLabels: {
+                      'app.kubernetes.io/name': 'metrics-server',
+                      'app.kubernetes.io/component': 'metrics-server',
+                    } + cfg.commonLabels,
+                  },
+                  namespaces: [cfg.namespace],
+                  topologyKey: 'kubernetes.io/hostname',
+                },
+              ],
+            },
+          },
+          containers: [
+            {
+              args: [
+                '--secure-port=10250',
+                '--kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname',
+                '--kubelet-use-node-status-port',
+                '--metric-resolution=15s',
+                '--kubelet-certificate-authority=/etc/tls/kubelet-serving-ca-bundle/ca-bundle.crt',
+                '--kubelet-client-certificate=/etc/tls/metrics-client-certs/tls.crt',
+                '--kubelet-client-key=/etc/tls/metrics-client-certs/tls.key',
+                '--tls-cert-file=/etc/tls/private/tls.crt',
+                '--tls-private-key-file=/etc/tls/private/tls.key',
+                '--tls-cipher-suites=' + cfg.tlsCipherSuites,
+              ],
+              image: 'registry.k8s.io/metrics-server/metrics-server:v0.6.3',
+              imagePullPolicy: 'IfNotPresent',
+              livenessProbe: {
+                failureThreshold: 3,
+                httpGet: {
+                  path: '/livez',
+                  port: 'https',
+                  scheme: 'HTTPS',
+                },
+                periodSeconds: 10,
+              },
+              name: 'metrics-server',
+              ports: [
+                {
+                  containerPort: 10250,
+                  name: 'https',
+                  protocol: 'TCP',
+                },
+              ],
+              readinessProbe: {
+                failureThreshold: 3,
+                httpGet: {
+                  path: '/readyz',
+                  port: 'https',
+                  scheme: 'HTTPS',
+                },
+                initialDelaySeconds: 20,
+                periodSeconds: 10,
+              },
+              resources: {
+                requests: {
+                  cpu: '100m',
+                  memory: '200Mi',
```

Comment:
We'd need to adjust these numbers following https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#resources-and-limits